src/adapter/scriptSkipper.ts

```diff
@@ -0,0 +1,86 @@
+import cdp from '../cdp/api';
+
+export class ScriptSkipper {
+  private _userSkipPatterns: string[];
+  private _nonNodeInternalRegex: string = "";
+
+  // filtering node internals
+  private _nodeInternalsRegex: string = "";
+  skipAllNodeInternals: boolean = false;
+
+  private _isUrlSkippedMap = new Map<string, boolean>();
+
+  private _blackboxedUrls: string[] = [];
+  private blackboxSender?: (params: cdp.Debugger.SetBlackboxPatternsParams) => Promise<cdp.Debugger.SetBlackboxPatternsResult | undefined>;
+
+  constructor(skipPatterns: string[]) {
+    this._userSkipPatterns = skipPatterns;
+    this._preprocessNodeInternals();
+    this._setRegexForNonNodeInternals();
+  }
+
+  private _preprocessNodeInternals(): void {
+    let nodeInternalRegex = new RegExp("^<node_internals>\/(.*)$");
+    let nodeInternalPatterns = this._userSkipPatterns!.filter(pattern => pattern.includes("<node_internals>")).map( nodeInternal => {
+      return nodeInternalRegex.exec(nodeInternal)![1];
+    });
+    if (nodeInternalPatterns.length > 0) { // set variables to use to filter node internals when in module
+      if (nodeInternalPatterns.includes("**/*.js"))
+        this.skipAllNodeInternals = true; // this value is checked later at a later processing point
+      else
+        this._nodeInternalsRegex = this._createRegexString(nodeInternalPatterns);
+    }
+  }
+
+  private _setRegexForNonNodeInternals(): void {
+    var nonNodeInternalGlobs = this._userSkipPatterns.filter(pattern => !pattern.includes("<node_internals>"));
+    this._nonNodeInternalRegex += this._createRegexString(nonNodeInternalGlobs);
```

Comment:
Keeping regexes and strings and concatenating the string with no `|` looks suspicious but I haven't looked at it in depth. Also, I am not sure why you are keeping the "non node internal" separate from the "node internal" patterns.

I know you want to get this into master so you don't have to resolve conflicts every day, I am ok merging this as a work in progress. I will wait to do an in depth review, I am guessing it will change when dynamic updating is implemented. Or I can review it more before that if you want.

Reply:
I'm joining the patterns in the _createRegexString function, which concatenates them with a `|`. For the "non node internal" separation, it's mostly because for Node, I'm waiting until a Node target is registered so that we get the node target from the process itself. I guess we can concatenate the two lists into a single regex, if that'd be better. 

Yes, I'd like to get this merged as a work in process -- but I'd also like to clean this up as much as possible 😅. But here is a todo list of changes I'd like to have in a second PR:
- working unit tests
- working integration tests
- error handling (don't show stacktrace when error occurs)
- browser launch/attach support
- add skipfiles to intellisense
- dynamic toggling of skip files